### PR TITLE
enable more test for SQL

### PR DESCRIPTION
**Pull request by snopoke**
changes to:
- submission errors
- user registration

@benrudolph 

cc @calellowitz @biyeun

### CODE REVIEW

File: corehq/form_processor/backends/couch/processor.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from casexml.apps.case.xform import get_case_updates
 from corehq.form_processor.exceptions import CaseNotFound
 from couchforms.util import process_xform, deprecation_type
-from couchforms.models import XFormInstance, XFormDeprecated, XFormDuplicate, doc_types, XFormError
+from couchforms.models import XFormInstance, XFormDeprecated, XFormDuplicate, doc_types, XFormError, SubmissionErrorLog
```
Comment (dimagimon) [inline corehq/form_processor/backends/couch/processor.py]: E501 line too long (119 > 115 characters)

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from django.db.models import F
```
Comment (dimagimon) [inline corehq/form_processor/backends/sql/dbaccessors.py]: F401 'F' imported but unused

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        order = 'received_on'
+        if recent_first:
+            order = '-{}'.format(order)
+
+        return XFormInstanceSQL.objects.filter(
+            domain=domain,
+            state=state
+        ).order_by(order)[0:limit]
+
```
Comment (dimagimon) [inline corehq/form_processor/backends/sql/dbaccessors.py]: W391 blank line at end of file

**Comment by benrudolph** [comment]
nicely done, test failure looks real:
```
======================================================================
ERROR: test_wrong_doc_type (couchforms.tests.test_duplicates.DuplicateFormTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/test_utils.py", line 168, in inner
    return helper(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/test_utils.py", line 141, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/test_utils.py", line 156, in call_with_settings
    fn(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/test_utils.py", line 137, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/couchforms/tests/test_duplicates.py", line 45, in test_wrong_doc_type
    self.interface.xform_model.archive(xform1)
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 248, in archive
    xform_archived.send(sender="form_processor", xform=self)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 198, in send
    response = receiver(signal=self, sender=sender, **named)
  File "corehq/ex-submodules/casexml/apps/case/signals.py", line 12, in rebuild_form_cases
    case_ids = get_case_ids_from_form(xform)
  File "corehq/ex-submodules/casexml/apps/case/xform.py", line 289, in get_case_ids_from_form
    return set(cu.id for cu in get_case_updates(xform))
  File "corehq/ex-submodules/casexml/apps/case/xform.py", line 285, in get_case_updates
    return [case_update_from_block(cb) for cb in extract_case_blocks(xform)]
  File "corehq/ex-submodules/casexml/apps/case/xform.py", line 248, in extract_case_blocks
    form = doc.form_data
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 127, in __call__
    value = self.func( *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 191, in form_data
    xml = self.get_xml()
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 232, in get_xml
    return self.get_attachment('form.xml')
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 53, in get_attachment
    attachment = self.get_attachment_meta(attachment_name)
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 66, in get_attachment_meta
    return _get_attachment_from_list(getattr(self, list_attr))
  File "/home/travis/build/dimagi/commcare-hq/corehq/form_processor/models.py", line 60, in _get_attachment_from_list
    for attachment in attachments:
TypeError: 'NoneType' object is not iterable
```

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by benrudolph** [comment]
woot. chug chug :train:

**Pull request merged**